pkg/routeagent/controllers/route/route.go

```diff
@@ -280,23 +522,38 @@ func (r *Controller) handleRemovedCluster(obj interface{}) {
 	// ideally we should attempt to remove all routes if the endpoint matches our cluster ID
 }
 
+func (r *Controller) handleRemovedPod(obj interface{}) {
+	klog.V(6).Infof("Removing podIP in route controller %v", obj)
+	pod := obj.(*k8sv1.Pod)
+
+	if r.remoteVTEPs.Contains(pod.Status.HostIP) {
+		r.remoteVTEPs.Delete(pod.Status.HostIP)
+		if r.isGatewayNode {
+			ret := r.vxlanDevice.DelFDB(net.ParseIP(pod.Status.PodIP), "00:00:00:00:00:00")
```

Comment:
`vxlanDevice` could be nil here if an _Endpoint_ has not yet been observed.

Reply:
Done